src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -1993,16 +1993,25 @@ internal static string ConcatenateStringPathArguments(CommandElementAst stringAs
             CompletionContext context,
             Dictionary<string, AstParameterArgumentPair> boundArguments = null)
         {
-            if (string.IsNullOrEmpty(commandName))
+            var parameterName = parameter.Name;
+
```

Comment:
To make the check and the `parameterFullName` logic simpler here, you could maybe do:

```csharp
string actualCommandName = string.IsNullOrEmpty(commandName)
    ? commandAst.GetCommandName()
    : commandName;

if (string.IsNullOrEmpty(actualCommandName))
{
    return;
}

string parameterFullName = actualCommandName + ":" + parameterName;
```

Reply:
That flows much better to me too. Updated.